cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/AbstractDatasetDefinition.java

```diff
@@ -44,4 +48,23 @@ protected AbstractDatasetDefinition(String name) {
   public String getName() {
     return name;
   }
+
+  /**
+   * Reconfigure a dataset instance. Delegates to {@link Reconfigurable#reconfigure} if the dataset definition
+   * implements that interface, and delegates to {@link #configure} otherwise.
+   * @param def a dataset definition
```

Comment:
well it is a dataset instance that is reconfigured. The definition is the class that performs this operation. I changed it to "the dataset definition that will perform the (re)configure"